runway/templates/k8s-flux-repo/flux.tf/main.tf

```diff
@@ -0,0 +1,468 @@
+
+terraform {
+  backend "s3" {
+    key = "eks-flux.tfstate"
+  }
+}
+
+locals {
+  cluster_name = "k8s-${terraform.workspace}"
+  flux_repository = "flux-${terraform.workspace}"
+  flux_namespace = "flux"
+  flux_service_account = "flux"
+  flux_version = "1.15.0"
+}
+
+data "aws_eks_cluster" "cluster" {
+  name = "${local.cluster_name}"
+}
+
+data "aws_eks_cluster_auth" "cluster_auth" {
+  name = "${data.aws_eks_cluster.cluster.id}"
+}
+
+data "aws_ssm_parameter" "oidc_iam_provider_cluster_url" {
+  name = "/${local.cluster_name}/oidc-iam-provider-cluster-url"
+}
+data "aws_ssm_parameter" "oidc_iam_provider_cluster_arn" {
+  name = "/${local.cluster_name}/oidc-iam-provider-cluster-arn"
+}
+
+provider "kubernetes" {
+  host = "${data.aws_eks_cluster.cluster.endpoint}"
+  cluster_ca_certificate = "${base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)}"
+  token = "${data.aws_eks_cluster_auth.cluster_auth.token}"
+  load_config_file = false
+  version = "1.9"
+}
+
+provider "aws" {
+  version = "~> 2.0"
+  region  = "us-east-1"
+}
+
+resource "kubernetes_namespace" "flux" {
+  metadata {
+    name = "${local.flux_namespace}"
+  }
+}
+resource "aws_codecommit_repository" "flux_repository" {
+  repository_name = "${local.flux_repository}"
+}
+
+data "aws_iam_policy_document" "flux_policy" {
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "codecommit:*"
+    ]
+    resources = [
+      "${aws_codecommit_repository.flux_repository.arn}"
+    ]
+  }
+
+}
+
+data "aws_iam_policy_document" "flux_service_account_assume_role_policy" {
+  statement {
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+    effect  = "Allow"
+
+    condition {
+      test     = "StringEquals"
+      variable = "${replace(data.aws_ssm_parameter.oidc_iam_provider_cluster_url.value, "https://", "")}:sub"
+      values   = ["system:serviceaccount:${local.flux_namespace}:${local.flux_service_account}"]
+    }
+
+    principals {
+      identifiers = ["${data.aws_ssm_parameter.oidc_iam_provider_cluster_arn.value}"]
+      type        = "Federated"
+    }
+  }
+}
+
+resource "aws_iam_role" "flux_service_account" {
+  name               = "${local.flux_service_account}"
+  assume_role_policy = "${data.aws_iam_policy_document.flux_service_account_assume_role_policy.json}"
+}
+
+resource "aws_iam_role_policy" "flux_service_account" {
+  role   = "${aws_iam_role.flux_service_account.id}"
+  policy = "${data.aws_iam_policy_document.flux_policy.json}"
+}
+
+resource "aws_ecr_repository" "flux_with_awscli" {
+  name                 = "${local.flux_repository}"
+  image_tag_mutability = "MUTABLE"
+  image_scanning_configuration {
+    scan_on_push = true
+  }
+}
+
+resource "null_resource" "flux_image" {
+
+  depends_on = [
+    "aws_ecr_repository.flux_with_awscli",
+  ]
+
+  provisioner "local-exec" {
```

Comment:
Would be *really* nice to replace this with something like https://github.com/terraform-providers/terraform-provider-docker/pull/133

Without that, I _think_ our current standard to keep this cross-platform is to move this functionality to a python file, and note the python dependency in the README (not sure if the existing `k8s-tf-repo` gen-sample notes that dependency, if not we should add it there too)